sessionctx/variable/variable.go

```diff
@@ -585,6 +591,21 @@ func init() {
 	}
 }
 
+// TruncateMaxAllowedPacket is used to truncate the value of max_allowed_packet to be a multiple of 1024,
+// nonmultiples are rounded down to the nearest multiple.
+func TruncateMaxAllowedPacket(sctx *stmtctx.StatementContext, val *string) error {
```

Comment:
IMHO it's better to define the function as:

```
func TruncateMaxAllowedPacket(sctx *stmtctx.StatementContext, val string) (uint, error)
```
For 2 reasons:

1. The 'side effect'(modification of income parameter) can be avoided.
2. You don't need to parse a string and convert the integer back to string.